integration-tests/scaffolding/helpers.ts

```diff
@@ -1,22 +1,166 @@
+    source: DevAccounts.Alice,
+    context
+});
+
+export class EventError extends Error {
+    name: string = '';
+    message: string = '';
+    stack?: string = '';
+    section?: string = '';
+    rawError: DispatchError;
+
+    constructor(source: DispatchError) {
+        super();
+
+        if (source) {
+            if (source.isModule) {
+                const decoded = source.registry.findMetaError(source.asModule);
+                this.name = decoded.name;
+                this.message = decoded.docs.join(' ');
+                this.section = decoded.section;
+            } else {
+                this.name = source.type;
+                this.message = source.type;
+                this.section = '';
+            }
+        }
+        this.rawError = source;
+    }
+
+    public toString() {
+        return `${this.section}.${this.name}: ${this.message}`;
+    }
+}
+
+export type EventMap = { [key: string]: Event }
+export type Sr25519Signature = { Sr25519: `0x${string}` }
+
+interface BeginEndBalances {
+    beginBalance: bigint;
+    endBalance?: bigint;
+}
+let balancesMap = new Map<string, Map<string, BeginEndBalances>>;
+let doTxAccounting = false;
+export let txAccountingHook = async (api, context?): Promise<void> => { };
+
+if (process.env.ENABLE_TX_ACCOUNTING === 'true' || process.env.ENABLE_TX_ACCOUNTING === '1') {
+    doTxAccounting = true;
+    txAccountingHook = async (api, context?) => showTotalCost(api, context);
+}
+
+export function parseResult() {
     return pipe(
-        map((result: ISubmittableResult) => result.events.reduce((acc, {event}) => { acc[eventKey(event)] = event.data; return acc}, {})),
-    )
+        tap((result: ISubmittableResult) => {
+            if (result.dispatchError) {
+                let err: EventError;
+
+                switch (result.dispatchError.type) {
+                    case 'Module': {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+
+                    default: {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+                }
+
+                console.log(err.toString());
+                throw err;
+            }
+        }),
+        map((result: ISubmittableResult) => result.events.reduce((acc, { event }) => { acc[eventKey(event)] = event.data; return acc }, {})),
+        // tap((events) => console.log(events)),
+    );
 }
 
-export function eventKey(event:Event): string {
+export function eventKey(event: Event): string {
     return `${event.section}.${event.method}`;
 }
 
 export function signPayloadSr25519(keys: KeyringPair, data: Codec): Sr25519Signature {
-    return {Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a())))}
-}
\ No newline at end of file
+    return { Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a()))) }
+}
+
+export function createAccount(name: string = 'first pair') {
```

Comment:
Maybe we can call this`createKeypair` so as not to be confused with creating an MSA account or a token account.
No strong opinion.

Reply:
I like it. After discussing some more with @rlaferla , I'd like to get this in as an initial overhaul & create a followup ticket for some more enhancement/standardization to the integration test framework. I intend to clean up the names some more there.